### DISCUSSION TITLE
Ability to couple with other models

### DISCUSSION THREAD

**Discussion by suvayu** [TulipaEnergy/TulipaEnergyModel.jl#138] [Ideas]
### Summary

In one of our other projects (DEMOSES), where COMPETES was supposed to be used, they decided not to go with it.  Instead they want to use a simple dispatch model and further explore flexibility options and market design on top of it.

They are hoping they can use Tulipa after our end-of-the-year release (since it will be the successor of COMPETES).

### Basic Idea

Their general approach is to incorporate smaller (regional?) models by coupling them.  AFAIU, the general consensus on the approach is to solve in a time window and move the window forward.  This is to limit the computational requirements, also it's closer to the real world since it relies on less coordination between parties.

### Question

Is this something within the scope of Tulipa?  If so, at what stage?  I also asked the PI to schedule a discussion with 'el patrón' :wink: 

If this is in scope, my personal opinion is we should engage early, so that we can iterate on implementations faster.

**FYI:** @sjvrijn is the DEMOSES lead from our side

**Comment by datejada** [comment]
Having a prototype that runs with the basic features is feasible by the end of the year. Even if the matrices for flexible periods are still being prepared by then, we can always run it hourly, especially if they want to run small case studies to avoid computational problems. I think it is a good opportunity to test Tulipa and start using it :)

@g-moralesespana, you commented that we need to coordinate with the DEMOSES project, didn't you? So, it might make sense to try to see their needs and if Tulipa can be a feasible option for them in time and level of detail.

**Comment by clizbe** [comment]
It's been decided that DEMOSES will use Tulipa.
